### ISSUE TITLE
[Surf's Up scenario issues] reinforcements info in science DB does not match comms report

### ISSUE THREAD

**Issue by muerteFR** [daid/EmptyEpsilon#2266]
~~**1.**  I don't know if it's normal that the reinforcements offered in a station are not the same as those displayed in the DB information.~~

![Owen](https://github.com/user-attachments/assets/b2cab634-9176-441d-9ba5-3a5600bfa5c7)

![Owen 2](https://github.com/user-attachments/assets/dc5de335-9b1c-4d09-bef9-b1e302245e23)

**2.** When the enemy base has defense platforms, if the base is destroyed, the defense platforms will self destruct after the rest of the enemy ships are destroyed. The message that should appear in Shiplog after about 60s appears well with Xansta and BlueShadow but not with me despite several tests.
Xansta
![4](https://github.com/user-attachments/assets/07d825bb-9262-4f3f-a2a8-3e97f21cfa36)

muerte
![1](https://github.com/user-attachments/assets/1599ca73-3248-43b1-95e4-e0b22fac4c7d)
![3](https://github.com/user-attachments/assets/03e7e0d5-2a95-4944-9592-aaf26ba5acc5)
![2](https://github.com/user-attachments/assets/f77a227a-90f1-4aff-ad72-da9c708dec66)

Edit : BlueShadows tried in DE and it works. The problem comes from me.
![error](https://github.com/user-attachments/assets/ccdbf2b3-5c49-4ffe-a467-ddc05faf75da)

**Comment by Xansta** [comment]
For item 1 regarding the services listed in the database not matching what shows up in relay: The station friendliness is one factor that determines whether a service is available or not, particularly when it comes to reinforcements. If you take a look at this table in the communications utility (comms_scenario_utility.lua):
```lua
	local reinforcement_info = {
		{desc = _("stationAssist-comms","Adder MK3"),			template = "Adder MK3",		threshold = 20,	cost = math.ceil(comms_target.comms_data.service_cost.amk3_reinforcements),		avail = comms_target.comms_data.service_available.amk3_reinforcements},
		{desc = _("stationAssist-comms","MU52 Hornet"),			template = "MU52 Hornet",	threshold = 50,	cost = math.ceil(comms_target.comms_data.service_cost.hornet_reinforcements),	avail = comms_target.comms_data.service_available.hornet_reinforcements},
		{desc = _("stationAssist-comms","Standard Adder MK5"),	template = "Adder MK5",		threshold = 0,	cost = math.ceil(comms_target.comms_data.service_cost.reinforcements),			avail = comms_target.comms_data.service_available.reinforcements},
		{desc = _("stationAssist-comms","Adder MK8"),			template = "Adder MK8",		threshold = 33,	cost = math.ceil(comms_target.comms_data.service_cost.amk8_reinforcements),		avail = comms_target.comms_data.service_available.amk8_reinforcements},
		{desc = _("stationAssist-comms","Phobos T3"),			template = "Phobos T3",		threshold = 66,	cost = math.ceil(comms_target.comms_data.service_cost.phobos_reinforcements),	avail = comms_target.comms_data.service_available.phobos_reinforcements},
		{desc = _("stationAssist-comms","Stalker R7"),			template = "Stalker R7",	threshold = 70,	cost = math.ceil(comms_target.comms_data.service_cost.stalker_reinforcements),	avail = comms_target.comms_data.service_available.stalker_reinforcements},
	}
```
...you can see a description (desc), template, threshold, cost and availability (avail) items for each reinforcement. Not only does the availability have to be true, the station "friendlyness" value has to be above the threshold value. In some cases, the friendliness of a station changes with circumstances that occur during the mission. So, while the service may not be offered by the station at first, it may be offered later if the station friendliness improves. Similarly, the station may have offered the service at first, but the station becomes less friendly during the mission (maybe the players were not very careful while spraying around their missiles) and the service stops being offered.

The database just shows that the service is possible at the station. That service may or may not be available when the Relay officer tries to request the service.

If this is an issue, I can just stop adding stations to the database so that the player's minds are less confused about the dynamics of the information given to them. For now, though, I plan to leave it in as it is.

**Comment by muerteFR** [comment]
Ok, I understand better. This poses no problem for me, on the contrary, I find it really good to enrich the DB during a mission/scenario. I noted this, as I said, without knowing if it was normal/intended or not, in order to bring you up to date as much as possible. And there will be other things noted on my part which will surely be in the same case.

**Comment by muerteFR** [comment]
For item 2, this doesn't matter much. It's probably something in my translation that causes this sentence not to appear, even when I don't put a translation on it.
This "issue" can be closed.

**Comment by muerteFR** [comment]
This "issue" can be closed.

**Comment by Xansta** [comment]
For item 2 where you are trying to see the "Defense platform self destruct activated" message, I think this may be an issue with the difference between two Empty Epsilon functions: onDestroyed() and onDestruction(). Surf's Up! uses onDestruction for the enemy station:
```lua
enemy_station:onDestruction(baseDestroyed)
```
The onDestroyed() function is triggered under any circumstance where an object is destroyed. In the context of the Empty Epsilon Lua code, "destroyed" means removed as a valid object from the virtual Empty Epsilon world.

In this case, we're talking about the enemy station that the player must destroy to complete the mission goal for the wave. The onDestruction() function is triggered only when the object is destroyed by taking damage. So, if you tested this scenario by using the GM screen and selecting the station and pressing the delete key, the function that sends the "Defense platform self destruct activated" message won't be activated because the enemy base was not destroyed by taking damage but by the omnipotent hand of the GM.
```lua
function baseDestroyed(self,instigator)
	string.format("")
	sub_wave_time = getScenarioTime() + 7
	local destruct_warning = false
	for i,enemy in ipairs(enemy_list) do
		if enemy ~= nil and enemy:isValid() and enemy:getTypeName() == "Defense platform" then
			destruct_warning = true
			break
		end
	end
	if destruct_warning then
		for i,p in ipairs(getActivePlayerShips()) do
			p:addToShipLog(_("shipLog","Defense platform self destruct activated."),"Red")
		end
	end
end
```

Before I investigate further, please describe how you destroyed the enemy station in your testing.

**Comment by muerteFR** [comment]
> Pour l'élément 2 où vous essayez de voir le message « Autodestruction de la plate-forme de défense activée », je pense que cela peut être un problème avec la différence entre deux fonctions Empty Epsilon : onDestroyed() et onDestruction(). Surf's Up! utilise onDestruction pour la station ennemie :
> 
> ```lua
> enemy_station:onDestruction(baseDestroyed)
> ```
> 
> La fonction onDestroyed() est déclenchée dans toutes les circonstances où un objet est détruit. Dans le contexte du code Lua Empty Epsilon, « détruit » signifie supprimé en tant qu'objet valide du monde virtuel Empty Epsilon.
> 
> Dans ce cas, nous parlons de la station ennemie que le joueur doit détruire pour compléter l'objectif de la mission de la vague. La fonction onDestruction() est déclenchée uniquement lorsque l'objet est détruit en subissant des dégâts. Donc, si vous avez testé ce scénario en utilisant l'écran du MJ et en sélectionnant la station et en appuyant sur la touche Suppr, la fonction qui envoie le message "Autodestruction de la plateforme de défense activée" ne sera pas activée car la base ennemie n'a pas été détruite en subissant des dégâts mais par la main toute-puissante du MJ.
> 
> ```lua
> function baseDestroyed(self,instigator)
> 	string.format("")
> 	sub_wave_time = getScenarioTime() + 7
> 	local destruct_warning = false
> 	for i,enemy in ipairs(enemy_list) do
> 		if enemy ~= nil and enemy:isValid() and enemy:getTypeName() == "Defense platform" then
> 			destruct_warning = true
> 			break
> 		end
> 	end
> 	if destruct_warning then
> 		for i,p in ipairs(getActivePlayerShips()) do
> 			p:addToShipLog(_("shipLog","Defense platform self destruct activated."),"Red")
> 		end
> 	end
> end
> ```
> 
> Avant d'enquêter davantage, veuillez décrire comment vous avez détruit la station ennemie lors de vos tests.

Indeed, I just tried by destroying it with missiles and this time it appears.
I had to do it during my first test but not after that. Sorry for this waste of time.

**Comment by muerteFR** [comment]
> For item 2 where you are trying to see the "Defense platform self destruct activated" message, I think this may be an issue with the difference between two Empty Epsilon functions: onDestroyed() and onDestruction(). Surf's Up! uses onDestruction for the enemy station:
> 
> ```lua
> enemy_station:onDestruction(baseDestroyed)
> ```
> 
> The onDestroyed() function is triggered under any circumstance where an object is destroyed. In the context of the Empty Epsilon Lua code, "destroyed" means removed as a valid object from the virtual Empty Epsilon world.
> 
> In this case, we're talking about the enemy station that the player must destroy to complete the mission goal for the wave. The onDestruction() function is triggered only when the object is destroyed by taking damage. So, if you tested this scenario by using the GM screen and selecting the station and pressing the delete key, the function that sends the "Defense platform self destruct activated" message won't be activated because the enemy base was not destroyed by taking damage but by the omnipotent hand of the GM.
> 
> ```lua
> function baseDestroyed(self,instigator)
> 	string.format("")
> 	sub_wave_time = getScenarioTime() + 7
> 	local destruct_warning = false
> 	for i,enemy in ipairs(enemy_list) do
> 		if enemy ~= nil and enemy:isValid() and enemy:getTypeName() == "Defense platform" then
> 			destruct_warning = true
> 			break
> 		end
> 	end
> 	if destruct_warning then
> 		for i,p in ipairs(getActivePlayerShips()) do
> 			p:addToShipLog(_("shipLog","Defense platform self destruct activated."),"Red")
> 		end
> 	end
> end
> ```
> 
> Before I investigate further, please describe how you destroyed the enemy station in your testing.

Indeed, I just tried by destroying it with missiles and this time it appears.
I had to do it during my first test but not after that. Sorry for this waste of time.
